src/index.js

```diff
@@ -1,63 +1,25 @@
-import Toolbar from './toolbar';
-import EditBtn from './editbtn';
-import Experiments from './experiments';
-import Utils from './utils';
-import Version from './version';
 import Share from './share';
-
-const PRISMIC_ENDPOINT = window.prismic && window.prismic.endpoint;
-let START_EXPERIMENT = () => {};
-
-const config = (() => {
-  const matches = PRISMIC_ENDPOINT && PRISMIC_ENDPOINT.match(new RegExp('(https?://([^/]*))'));
-  if (matches && matches[1] && matches[2]) {
-    const baseURL = matches[1].replace(/\.cdn\.prismic\.io/, '.prismic.io');
-    const editorTab = matches[2].replace(/\.cdn\.prismic\.io/, '.prismic.io');
-    const location = {
-      origin: window.location.origin,
-      hash: window.location.hash,
-      pathname: window.location.pathname,
-      search: window.location.search,
-    };
-    return { baseURL, editorTab, location };
-  }
-  return null;
-})();
-
-const setupToolbar = Utils.debounce(() => Toolbar.setup(), 500, true);
-
-function setupEditButton() {
-  if (config) {
-    EditBtn.setup(config);
-  }
-}
-
-function startExperiment(expId) {
-  if (expId) {
-    START_EXPERIMENT = () => Experiments.start(expId);
-  }
-}
-
-document.addEventListener('DOMContentLoaded', () => {
-  if (config) {
-    Share.listen(config, () => {
-      START_EXPERIMENT();
+import Utils from './utils';
+import Config from './config';
+import Globals, { startExp, setupToolbar, setupEditButton } from './globals';
+
+(async () => {
+  // Globals
+  window.prismic = Globals;
+
+  // Polyfills
+  const features = [];
+  if (!window.Promise) features.push('Promise');
+  if (!window.fetch) features.push('fetch');
+  if (features.length) await Utils.script(`https://cdn.polyfill.io/v2/polyfill.min.js?features=${features.join(',')}&flags=gated,always`);
+
+  // Setup
+  setTimeout(() => {
```

Comment:
what's that timeout? why don't you use DOMContentLoaded in addition to the document.readyState === "complete"
Seems less hacky